src/pmacfiltercontrol/pmacFilterControlWrapper.py

```diff
@@ -1021,6 +1021,10 @@ def _set_filter_set(self, filter_set_num: int) -> None:
 
         self.filter_set_rbv.set(filter_set_num)
 
+        if self.attenuation.get() == 15:
+            self.attenuation.set(14)
+        self.attenuation.set(15)
```

Comment:
This is a bit horrible. Maybe it could you call `_set_manual_attenuation` directly to make sure the demand is sent?